salt/modules/cassandra_cql.py

```diff
@@ -0,0 +1,585 @@
+
+    .. code-block:: bash
+
+        salt 'minion1' cassandra.create_keyspace keyspace=newkeyspace
+
+        salt 'minion1' cassandra.create_keyspace keyspace=newkeyspace replication_strategy=NetworkTopologyStrategy \
+        replication_datacenters='{"datacenter_1": 3, "datacenter_2": 2}'
+    '''
+    existing_keyspace = keyspace_exists(keyspace, contact_points, port)
+    if not existing_keyspace:
+        # Add the strategy, replication_factor, etc.
+        replication_map = {
+            'class': replication_strategy
+        }
+        if replication_datacenters:
+            if isinstance(replication_datacenters, basestring):
+                try:
+                    replication_datacenter_map = json.loads(replication_datacenters)
+                    replication_map.update(**replication_datacenter_map)
+                except:
+                    log.error("Could not load json replication_datacenters.")
+                    return False
+            else:
+                replication_map.update(**replication_datacenters)
+        else:
+            replication_map['replication_factor'] = replication_factor
+        query = "create keyspace %s with replication = %s and durable_writes = true;" % (keyspace, replication_map)
+        _query(query, contact_points, cql_user, cql_pass, port)
+        return keyspace_exists(keyspace, contact_points, port, cql_user, cql_pass)
+    else:
+        return existing_keyspace
+
+
+def drop_keyspace(keyspace, contact_points=None, port=None, cql_user=None, cql_pass=None):
+    '''
+    Drop a keyspace if it exists in a Cassandra cluster.
+
+    :param keyspace        The keyspace to drop.
+    :type  keyspace:       str
+    :param contact_points: The Cassandra cluster addresses, can either be a string or a list of IPs.
+    :type  contact_points: str | list[str]
+    :param cql_user:       The Cassandra user if authentication is turned on.
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :return:               The info for the keyspace or False if it does not exist.
+    :rtype:                dict
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt 'minion1' cassandra.drop_keyspace keyspace=test
+
+        salt 'minion1' cassandra.drop_keyspace keyspace=test contact_points=minion1
+    '''
+    existing_keyspace = keyspace_exists(keyspace, contact_points, port)
+    if existing_keyspace:
+        query = 'drop keyspace %s' % keyspace
+        ret = _query(query, contact_points, port, cql_user, cql_pass)
+        # The drop keyspace query doesn't actually return anything if the query succeeds.
+        # So if ret is empty, then the drop worked. Hence returning True for the function.
+        return True if not ret else ret
+    else:
+        return True
```

Comment:
That does not looks nice. If you rely on booleans, where True (succeeded), then logically there should be False (failed) and information in logs why so. If you like to return the reason, then it is better raise an exception in this case.

In your case it is either potential bug or you're asking for complexity of type checking: because either your method always returns something (and therefore it cannot be called as: `if drop_keyspace(...):` because it won't work, or it is needed to check whether you return `True` or string of failure.